### PR TITLE
OD-325 [Fix] Not necessary fields are displayed when creating a Filter option

### PR DESCRIPTION
**Pull request by AndrRyaz**
@sofiiakvasnevska 
## Issue
OD-325 https://weboo.atlassian.net/jira/software/projects/OD/boards/9?selectedIssue=OD-325
## Description
- Fix logics, 'Number is between' and 'Enter value' after merging branches into repo.
- Added updated error check condition. 
- Added fixes to filter logic.
## Screenshots/screencasts
http://somup.com/crV23aobjY
## Backward compatibility
This change is fully backward compatible.
## Reviewers
@upplabs-alex-levchenko @ivandevupp

### CODE REVIEW

File: interface.html [hunk 1]
```diff
@@ -25,6 +25,9 @@
     </div>
   </div>
   <div class="settings-tab state future">
+    <div class="error-holder relations-error hidden">
+      <span class="text-danger">Please provide a value on required fields.</span>
```
Comment (tonytlwu) [inline interface.html:29]: Incorrect copy

image

File: interface.html [hunk 2]
```diff
@@ -25,6 +25,9 @@
     </div>
   </div>
   <div class="settings-tab state future">
+    <div class="error-holder relations-error hidden">
```
Comment (tonytlwu) [inline interface.html:28]: Where's the `relations-error` class used? I can't understand what it's for.

File: js/interface-lists.js [hunk 3]
```diff
@@ -459,26 +459,59 @@ function attahObservers() {
...
 
           widgetData.filterOptions.forEach(function(item) {
+            if (item.logic === 'between') {
+              filterFieldValues.push({
+                field: '#value-field-from-' + item.id,
+                value: item.value.from,
+                id: item.id,
+                valueType: item.valueType.from
+              });
+
```
Comment (tonytlwu) [inline js/interface-lists.js:469]:
```suggestion
```

File: templates/interface/filter-panels.interface.hbs [hunk 4]
```diff
@@ -121,13 +122,14 @@
...
               <option value="enter-value">Enter a value</option>
+              <option value="user-profile-data">User profile data</option>
             </select>
             <span class="icon fa fa-chevron-down"></span>
           </label>
         </div>
       </div>
 
-      <div class="form-group clearfix">
+      <div class="form-group clearfix" id="filter-value-between-{{id}}">
```
Comment (tonytlwu) [inline templates/interface/filter-panels.interface.hbs:132]: There are two elements on the page with the same ID. We can't have that.

**Review by tonytlwu — CHANGES_REQUESTED** [review]

File: interface.html (hunk 2 above)
Comment (AndrRyaz) [reply interface.html:28]: @tonytlwu 
We already have such a class
изображение

Comment (tonytlwu) [reply interface.html:28]: Thanks

File: interface.html (hunk 1 above)
Comment (AndrRyaz) [reply interface.html]: @tonytlwu fixed

File: js/interface-lists.js (hunk 3 above)
Comment (AndrRyaz) [reply js/interface-lists.js:469]: @tonytlwu fixed

File: templates/interface/filter-panels.interface.hbs (hunk 4 above)
Comment (AndrRyaz) [reply templates/interface/filter-panels.interface.hbs]: @tonytlwu fixed

**Review by tonytlwu — APPROVED** [review]
@sofiiakvasnevska This is ready for testing

Risk: High
Affected: Studio users configuring LFD filters
Verify: LFD filters can be correctly set up

Squash and merge: Hide unnecessary fields when configuring filters

**Review by tonytlwu — DISMISSED** [review]

**Pull request merged**